crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -115,6 +116,30 @@ pub(crate) async fn fetch_balance(
     Ok(convert_bal(b))
 }
 
+fn convert_obj(s: &sui_json_rpc_types::SuiObjectData) -> Object {
+    Object {
+        version: s.version.into(),
+        digest: s.digest.to_string(),
+        storage_rebate: s.storage_rebate,
+        address: SuiAddress::from_array(**s.object_id),
+        owner: s
+            .owner
+            .unwrap()
+            .get_owner_address()
+            .map(|x| SuiAddress::from_array(x.to_inner()))
+            .ok(),
+        bcs: Some(Base64::from(&bcs::to_bytes(&s.bcs).unwrap())), // TODO: is this correct?
+        previous_transaction: Some(s.previous_transaction.unwrap().to_string()),
+        kind: Some(match s.owner.unwrap() {
+            NativeOwner::AddressOwner(_) | NativeOwner::ObjectOwner(_) => ObjectKind::Owned,
```

Comment:
Ah, actually `NativeOwner::ObjectOwner` deserves its own `ObjectKind`, probably `ObjectKind::Child`.  

Confusingly when you transfer the object at `0xA` to object `0xB`, (i.e. transfer-to-object), `0xA`'s owner is `AddressOwner(0xB)`, but we should distinguish that from the case where `0xA` is a dynamic object field of `0xB`, in which case `0xA`'s owner will be `ObjectOwner(0xF)`, where `0xF` is the Field, (which has owner `ObjectOwner(0xB)`).

I'll put out a PR to update the draft schema for this as well.